src/components/reusables/Filter/Filter.tsx

```diff
@@ -0,0 +1,150 @@
+  applyButtonLabel,
+  resetButtonLabel,
+  icon,
+  value,
+  searchable,
+  modalView = false,
+  closeButtonAriaLabel = 'Close ' + label,
+  onApply,
+}: FilterProps) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [activeFilters, setActiveFilters] = useState<string[]>(value ?? []);
+  const [checkedFilters, setCheckedFilters] = useState<string[]>(value ?? []);
+  const [hasChanges, setHasChanges] = useState(false);
+  const filterButtonID = useId();
+
+  // Update selected values when there are external changes
+  const prevValue = usePrevious(value);
+  useEffect(() => {
+    if (value !== undefined && !arraysEqual(value, prevValue)) {
+      setActiveFilters(value);
+    }
+  }, [value]);
+
+  useEffect(() => {
+    if (!arraysEqual(activeFilters, checkedFilters)) {
+      setHasChanges(true);
+    } else {
+      setHasChanges(false);
+    }
+  }, [checkedFilters]);
+
+  const handleApply = () => {
+    onApply?.(checkedFilters);
+    setActiveFilters([...checkedFilters]);
+    handleOpenOrClose();
+    setHasChanges(false);
+  };
+
+  const handleReset = () => {
+    setCheckedFilters([]);
+  };
+
+  const handleOpenOrClose = () => {
+    if (isOpen) {
+      setIsOpen(false);
+    } else {
+      setCheckedFilters([...activeFilters]); // reset to active choices
+      setIsOpen(true);
+    }
+  };
+
+  const modalHeader = () => (
+    <div className={classes.modalHeader}>
+      <h3>{label}</h3>
+      <Button
+        variant='quiet'
+        color='secondary'
+        onClick={handleOpenOrClose}
+        icon={<XMarkIcon />}
+        aria-label={closeButtonAriaLabel}
+      />
+    </div>
+  );
+
+  return (
+    <Floatover
+      trigger={
+        <FilterButton
+          id={filterButtonID}
+          onClick={handleOpenOrClose}
+          iconLeft={icon}
+          isOpen={isOpen}
+          numActiveFilters={activeFilters.length}
+        >
+          {label}
+        </FilterButton>
+      }
+      isOpen={isOpen}
+      setIsOpen={handleOpenOrClose}
+      isModal={modalView}
+    >
+      <div
+        aria-labelledby={filterButtonID}
+        className={classes.popoverContent}
+      >
+        {modalView && modalHeader()}
+        <div className={cn(classes.optionSection, { [classes.modal]: modalView })}>
+          <OptionDisplay
+            options={options}
+            onValueChange={setCheckedFilters}
+            value={checkedFilters}
+            searchable={searchable}
+            compact={!modalView}
+          />
+        </div>
+        <div className={cn(classes.filterActions, { [classes.modal]: modalView })}>
+          <Button
+            className={classes.resetButton}
+            size={modalView ? 'medium' : 'small'}
+            variant='quiet'
+            fullWidth={false}
+            aria-disabled={checkedFilters.length === 0}
+            onClick={checkedFilters.length === 0 ? undefined : handleReset}
+          >
+            {resetButtonLabel}
+          </Button>
+          <Button
```

Comment:
Vet ikke om dette er en bug men på Zalando så applies filteret når man trykker utenfor filteret og den lukker seg mens i denne her så må man trykke på Bruk-knappen før filteret blir applied. Lurer bare på hvilken grunn det er til at vi applyer filteret kun når man trykker på Bruk-knappen? 😊

Reply:
Jeg tenkte vel egentlig at hvis vi skal ha en 'Apply' knapp så bør vel den brukes 😅
Så for meg at det å klikke utenfor boksen var som å klikke på 'escape'-knappen, som føles som en 'cancel'-funksjonalitet mere enn en 'apply'-funksjonalitet.

Reply:
For å konkretisere så støtter jeg å ha en dedikert bruk-knapp fordi det er sikkert brukere som vil se etter en slik knapp. Men jeg ser de fleste andre filtrene der ute applyer filtrene enten med en gang man trykker på checkboxen(syns dette er en suboptimal løsning da) eller når popoveren lukker seg. Jeg tror at folk vil være opplært i denne måten å bruke komponenten på fra andre nettsider (akkurat som jeg er) og ikke skjønne hvordan de skal få filteret til å fungere når de bruker det hos oss. Men vi må gjerne få Randi og Hanne til å brukerteste akkurat dette om du er uenig

Reply:
Hmm, sant.
Jeg ser at Zalando f.eks. applyer selv om man trykker på esc også så hvis det er kotymen så slipper vi iallefall å differensiere mellom de to.
Men kan høre med dem (og Lasse når jeg først skal spøøre ham om noe) bare for sikkerhets skyld.